api/storage_service.cc

```diff
@@ -673,6 +692,14 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
     });
 
     ss::describe_ring.set(r, [&ctx, &ss](std::unique_ptr<http::request> req) {
+        if (!req->param.exists("keyspace")) {
+            throw bad_param_exception("The keyspace param is not provided");
+        }
```

Comment:
no need to check the existence of `keyspace`, an `std::out_of_range` is thrown if we access it with `req->param["keyspace"]` and it is missing.

Reply:
I think it is much better to tell the user "The keyspace param is not provided" than telling  "std::out_of_range".

Reply:
but i am wondering how come we can run into this error. if the "keyspace" path is required. as this introduces extra complexity, if we can dispense with it, we don't have to.

Reply:
We have a version that accept no keyspace. 

After disable it, I saw
```
$ curl --silent "http://127.0.0.1:10000/storage_service/describe_ring" | jq
{
  "message": "Not found",
  "code": 404
}
```

This is pretty bad error message which supposes to tell the user the keyspace path parameter is missed, instead of not found.

Reply:
right. but do we plan to drop this API which does not accept keyspace? if we do, i agree that this change is necessary, and improves the user experience. actually i was asking the reasons for adding this API when preparing for working on  #16509 , see https://github.com/scylladb/scylladb/pull/17098#issuecomment-1920762366

Reply:
This one is is there for legacy reasons. It is very easy to use it in a wrong way. I think we should deprecate it one day.

Reply:
okay, will create a PR to drop it.